### PR TITLE
chore: Better Errors

### LINKED ISSUE #925: chore: Error Logic Cleanup
We utilize similar errors across the following modules: Clock, FeePay, and CW-Hooks. Additionally, it would be good to standardize the way we execute contracts with child context between Clock and CW-Hooks.

### PR DESCRIPTION
**Pull request by joelsmith-2019**
Close #925 

Refactor error logic and error messages across multiple in-house modules.

### CODE REVIEW

File: x/clock/types/errors.go [hunk 1]
```diff
@@ -2,21 +2,10 @@ package types
...
-	ErrContractRegisterNotAdmin  = errorsmod.Register(ModuleName, 3, "this address is not the contract admin, cannot register")
-	ErrContractNotAdmin          = errorsmod.Register(ModuleName, 4, "sender is not the contract admin")
-	ErrContractNotCreator        = errorsmod.Register(ModuleName, 5, "sender is not the contract creator")
-	ErrContractJailed            = errorsmod.Register(ModuleName, 6, "contract is jailed")
-	ErrContractNotJailed         = errorsmod.Register(ModuleName, 7, "contract is not jailed")
-	ErrContractAlreadyJailed     = errorsmod.Register(ModuleName, 8, "contract is already jailed")
-	ErrInvalidCWContract         = errorsmod.Register(ModuleName, 9, "invalid CosmWasm contract")
-	ErrOutOfGas                  = errorsmod.Register(ModuleName, 10, "contract execution ran out of gas")
-	ErrContractExecutionPanic    = errorsmod.Register(ModuleName, 11, "contract execution panicked")
+	ErrContractJailed        = errorsmod.Register(ModuleName, 6, "contract is jailed")
```
Comment (Reecepbcups) [inline x/clock/types/errors.go:8]: should this be `errorsmod.Register(ModuleName, 1, "contract is jailed")` instead? (int) or does the gloablerror throw this off?

File: interchaintest/module_cwhooks_test.go [hunk 2]
```diff
@@ -57,6 +57,25 @@ func TestJunoCwHooks(t *testing.T) {
 	require.Equal(t, user.FormattedAddress(), res.Data.DelegatorAddress)
 	require.Equal(t, fmt.Sprintf("%d.000000000000000000", stakeAmt), res.Data.Shares)
 
+	// HIGH GAS TEST
+	// Setup a high gas contract
+	_, contractAddr = helpers.SetupContract(t, ctx, juno, user.KeyName(), "contracts/juno_staking_hooks_high_gas_example.wasm", `{}`)
```
Comment (Reecepbcups) [inline interchaintest/module_cwhooks_test.go:62]: make sure to link to this contract source in the README of the interchainttest dir

File: x/clock/types/errors.go (hunk 1 above)
Comment (joelsmith-2019) [reply x/clock/types/errors.go:8]: Good catch! I forgot to update the error numbers for the clock module after migrating the majority of them to global errors.

File: interchaintest/module_cwhooks_test.go (hunk 2 above)
Comment (joelsmith-2019) [reply interchaintest/module_cwhooks_test.go:62]: Good catch as well, added the references for the clock testing as well!

**Review by Reecepbcups — APPROVED** [review]

**Pull request merged**